src/coreclr/jit/utils.cpp

```diff
@@ -4254,3 +4240,81 @@ bool CastFromDoubleOverflows(double fromValue, var_types toType)
     }
 }
 } // namespace CheckedOps
+
+template <size_t bufferSize>
+class Utf16String
+{
+private:
+    wchar_t  m_buffer[bufferSize];
+    wchar_t* m_pBuffer = nullptr;
+
+public:
+    Utf16String(const char* str)
+    {
+        int strBufferSize = MultiByteToWideChar(CP_UTF8, 0, str, -1, nullptr, 0);
+        if (strBufferSize == 0)
+        {
+            return;
+        }
+
+        if (strBufferSize > bufferSize)
+        {
+            m_pBuffer = new wchar_t[strBufferSize];
+        }
+        else
+        {
+            m_pBuffer = m_buffer;
+        }
+
+        if (MultiByteToWideChar(CP_UTF8, 0, str, -1, m_pBuffer, strBufferSize) == 0)
+        {
+            if (m_pBuffer != m_buffer)
+            {
+                delete[] m_pBuffer;
+            }
+
+            m_pBuffer = nullptr;
+        }
+    }
+
+    ~Utf16String()
+    {
+        if (m_pBuffer != m_buffer)
+        {
+            delete[] m_pBuffer;
+            m_pBuffer = nullptr;
+        }
+    }
+
+    const wchar_t* Result()
+    {
+        return m_pBuffer;
+    }
+};
+
+//------------------------------------------------------------------------
+// fopen_utf8: Open the file at the specified UTF8 path with the specified mode.
+//
+// Arguments:
+//   path - UTF8 path
+//   mode - UTF8 mode
+//
+// Returns:
+//    Opened file handle
+//
+FILE* fopen_utf8(const char* path, const char* mode)
+{
+#ifdef HOST_WINDOWS
+    Utf16String<256> pathWide(path);
```

Comment:
Paths on Windows can be longer than 256 characters

Reply:
There is a dynamic allocation inside `Utf16String` for the longer case

Reply:
`MAX_PATH` would be more appropriate here.

Reply:
I chose 256 because that is a 512 byte stack allocation, which was the largest size stack allocation I wanted to go with for the buffer.

Reply:
Okay. Given this is defined to only be on Windows _and_ there is a macro with a predefined size for the "common" path on the platform, using `MAX_PATH` in a variable called `pathWide` seems like the most appropriate to me. If you think an explicit 256 is clearer then feel free to go with it.

Reply:
I prefer the 256 simply because the value I picked here only coincidentally is close to `MAX_PATH` and wasn't intended to have any semantic relevance to its value (it seems like an outdated define nowadays given that the max is much higher).